packages/studio-ui/src/web/components/Layout/TopNav/cloud/Step2-workspace.tsx

```diff
@@ -0,0 +1,111 @@
+import axios from 'axios'
+import { lang } from 'botpress/shared'
+import { UnreachableCaseError } from 'common/errors'
+import React, { useEffect, useReducer } from 'react'
+import { connect } from 'react-redux'
+import { RootReducer } from '~/reducers'
+import { Workspace } from './types'
+import { WorkspaceSelector } from './workspace'
+
+interface OwnProps {
+  pat: string
+  onCompleted: (selectedWorkspaceId: string) => void
+}
+
+type StateProps = ReturnType<typeof mapStateToProps>
+type DispatchProps = typeof mapDispatchToProps
+type Props = DispatchProps & StateProps & OwnProps
+
+type State =
+  | { status: 'checking_bot_workspace' }
+  | {
+      status: 'fetching_available_workspaces'
+    }
+  | {
+      status: 'received_available_workspaces'
+      workspaces: Workspace[]
+    }
+  | {
+      status: 'saving'
+      selectedWorkspace: Workspace
+    }
+
+type Action =
+  | { type: 'fetch_available_workspaces' }
+  | { type: 'availableWorkspaces/received'; value: Workspace[] }
+  | { type: 'save/clicked'; selectedWorkspace: Workspace }
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'fetch_available_workspaces':
+      return { status: 'fetching_available_workspaces' }
+    case 'availableWorkspaces/received':
+      return { status: 'received_available_workspaces', workspaces: action.value }
+    case 'save/clicked':
+      return { status: 'saving', ...action }
+    default:
+      throw new UnreachableCaseError(action)
+  }
+}
+
+const WorkspaceForm = (props: Props): JSX.Element => {
+  const { bot, pat, onCompleted } = props
+
+  const [state, dispatch] = useReducer(reducer, { status: 'checking_bot_workspace' })
+
+  const { status } = state
+
+  useEffect(() => {
+    const ac = new AbortController()
+
+    const fetchWorkspaces = async () => {
+      const { data: workspaces } = await axios.get(`${window.CLOUD_CONTROLLER_ENDPOINT}/v1/workspaces`, {
+        signal: ac.signal,
+        headers: { Authorization: `bearer ${pat}` }
+      })
```

Comment:
That would involve moving the `CloudClient` to `common` and ensure it is compatible both in the browser and the backend. I don't think it's worth the effort now, since the browser only does light usage of the controller API. What do you think?

Reply:
I'll try moving the CloudClient to `common`

Reply:
Fixed in d1c06cc